recipes/mrtrix3/build.sh

```diff
@@ -0,0 +1,7 @@
+#!/usr/bin/env bash
+
+export EIGEN_CFLAGS=$(pkg-config --cflags eigen3)
+
+./configure -conda -nogui -noshared -verbose
```

Comment:
The `noshared` flag here could be the source of your OSX issues. On one side, we (extremely) prefer shared linkage in conda-forge and on the other side the error complains about a too large binary. Going to shared linkage should also reduce the size of the binary.